core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -161,49 +178,68 @@ private[chisel3] trait HasId extends InstanceId {
     // This could be called with user prefixes, ignore them
     noPrefix {
       suggested_seed = Some(seed)
-      this.suggestName(seed)
+      this.suggestNameInternal(seed)
     }
   }
 
+  private[chisel3] var _computedName: Option[Option[String]] = None
```

Comment:
I'm not super keen on adding this `var` because it increases the memory size of every `HasId` by 4-bytes and every named `Data` by 28-bytes. I took a different approach in https://github.com/chipsalliance/chisel3/pull/2561 to just recalculate the name and check which maybe we can do here.